From 45007a0525eb10a15003550a91cacc2f6e878765 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Sun, 4 Aug 2024 14:04:23 -0300
Subject: [PATCH] tools: Fix compilation with musl and gcc 14

lscss.c: In function 'is_sch_vfio':
lscss.c:392:20: error: implicit declaration of function 'basename' [-Wimplicit-function-declaration]
  392 |         if (strcmp(basename(driver_path), "vfio_ccw") == 0)

device.c: In function 'device_read_active_attrib':
device.c:426:45: error: implicit declaration of function 'basename'; did you mean 'rename'? [-Wimplicit-function-declaration]
  426 |                         value = misc_strdup(basename(link));
      |                                             ^~~~~~~~
      |                                             rename
---
 zconf/css/lscss.c      | 1 +
 zdev/src/ccw.c         | 1 +
 zdev/src/ccwgroup.c    | 1 +
 zdev/src/device.c      | 1 +
 zdev/src/generic_ccw.c | 1 +
 5 files changed, 5 insertions(+)

diff --git a/zconf/css/lscss.c b/zconf/css/lscss.c
index fd58bb3..e75b2b2 100644
--- a/zconf/css/lscss.c
+++ b/zconf/css/lscss.c
@@ -9,6 +9,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <libgen.h>
 
 #include "lib/ccw.h"
 #include "lib/util_base.h"
diff --git a/zdev/src/ccw.c b/zdev/src/ccw.c
index e8c40f9..3507049 100644
--- a/zdev/src/ccw.c
+++ b/zdev/src/ccw.c
@@ -8,6 +8,7 @@
  */
 
 #include <inttypes.h>
+#include <libgen.h>
 #include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>
diff --git a/zdev/src/ccwgroup.c b/zdev/src/ccwgroup.c
index 61e12cc..6a35d6c 100644
--- a/zdev/src/ccwgroup.c
+++ b/zdev/src/ccwgroup.c
@@ -10,6 +10,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <libgen.h>
 
 #include "lib/util_base.h"
 #include "lib/util_path.h"
diff --git a/zdev/src/device.c b/zdev/src/device.c
index 61647df..63767b6 100644
--- a/zdev/src/device.c
+++ b/zdev/src/device.c
@@ -11,6 +11,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
+#include <libgen.h>
 
 #include "lib/util_path.h"
 
diff --git a/zdev/src/generic_ccw.c b/zdev/src/generic_ccw.c
index c7f144a..637ccfc 100644
--- a/zdev/src/generic_ccw.c
+++ b/zdev/src/generic_ccw.c
@@ -8,6 +8,7 @@
  */
 
 #include <string.h>
+#include <libgen.h>
 
 #include "lib/util_path.h"
 
-- 
2.45.2

